packages/auth/src/__tests__/auth.test.ts

```diff
@@ -1,6 +1,91 @@
-import Auth from '../index'
+import {rest} from 'msw'
+import {setupServer} from 'msw/node'
+import cookie from '@skillrecordings/cookies'
+import {ACCESS_TOKEN_KEY, USER_KEY} from '@skillrecordings/config'
 
-test('can create an instance', () => {
+import Auth, {getAuthorizationHeader, getLocalUser} from '../index'
+
+export const cornDogUser = {
+  name: 'Corn Dog',
+}
+
+export const server = setupServer(
+  // we can fake the actual user tat gets loaded with msw 🎉
+  rest.get(`/api/users/current`, (req, res, ctx) => {
+    return res(ctx.json(cornDogUser))
+  }),
+)
+
+beforeAll(() => server.listen())
+afterAll(() => server.close())
+
+afterEach(() => {
+  localStorage.removeItem(ACCESS_TOKEN_KEY)
+  cookie.remove(ACCESS_TOKEN_KEY)
+})
+
+test('can create an instance of Auth', () => {
   const auth = new Auth()
   expect(auth).not.toBeNull()
 })
+
+test('read the token from auth instance', async () => {
+  const auth = new Auth()
+  const accessToken = '0f95fab36e7a'
+
+  const token = await auth.eggheadAuth.token.getToken(
+    `${process.env.NEXT_PUBLIC_REDIRECT_URI}#access_token=${accessToken}&token_type=bearer&expires_in=5184000`,
+  )
+
+  expect(token.accessToken).toBe(accessToken)
+})
+
+test('handle new session', async () => {
+  const auth = new Auth()
+  const accessToken = '0f95fab36e7a'
+  const expiresIn = 5184000
+  const url = `${process.env.NEXT_PUBLIC_REDIRECT_URI}#access_token=${accessToken}&token_type=bearer&expires_in=${expiresIn}`
+
+  // set the testURL in package.json to the redirect DOMAIN, then we can
+  // push state and get a new URL and simulate a valid login
+  //https://github.com/facebook/jest/issues/890#issuecomment-506926841
+  window.history.pushState({}, '', url)
+
+  const viewer = await auth.handleAuthentication()
```

Comment:
`handleAuthentication` is a cascade that does quite a lot internally and respond with a `Viewer` (user) instance. As long as what it does is return a `Viewer` we are ignoring all that internal complexity. Since it digs in and makes the API call to return the user, I'm satisfied.